corehq/apps/reports/standard/tableau.py

```diff
@@ -77,7 +85,8 @@ def tableau_visualization_ajax(request, domain):
     target_site = visualization_data.pop('target_site')
 
     # Authenticate
-    if not request.couch_user.can_view_tableau_viz(domain, visualization_data.pop('viz_id')):
+    if not request.couch_user.can_view_tableau_viz(domain, TableauVisualization.objects.get(
```

Comment:
not anything blocking but maybe `_authenticate_request` could be a more general `can_view_tableau_viz` method that accepts the domain and TableauVisualization as args so you can use it here too.